pyrealm/pmodel/two_leaf_irradience.py

```diff
@@ -0,0 +1,31 @@
+"""To do."""
+
+import numpy as np
+from numpy.typing import NDArray
+
+
+class TwoLeafIrradience:
+    """Calculates two-leaf irradience."""
+
+    def __init__(self, beta_angle: NDArray, PPFD: NDArray, LAI: NDArray, PATM: NDArray):
+
+        self.beta_angle: NDArray = beta_angle
+
+
+def beta_angle(lat: NDArray, d: NDArray, h: NDArray) -> NDArray:
+    """Calculates solar beta angle.
```

Comment:
The reference for this is page 554 of this:
https://onlinelibrary.wiley.com/doi/epdf/10.1111/j.1365-3040.1997.00094.x

I think `beta_angle` here is (or should be) solar elevation angle - which might make it part of the solar functionality rather than an irradiance specific thing. If we move to using one of the existing solar packages, we might get a more accurate model.

Reply:
I've added pysolar alternatives in the latest commit which calculate solar elevation directly from latitude, longitude and date time. However, they don't take array inputs natively so you need to use and enumerate operation, so there will be a speed accuracy trade.

The best pysolar solar elevation implementation includes atmospheric distortion, but they also have a faster/simpler version which is the same as the first version I wrote. Because the first version uses numpy vectorisation it may well be faster but I've included the pysolar fast version anyway.